lib/deployment.js

```diff
@@ -1,87 +1,67 @@
 "use strict";
 
+var utils = require("./utils");
+
 module.exports = function deployment(request) {
     return {
         list: function list(cb) {
-            request("/api/deployments/", function deploymentListCallback(err, response, body) {
-                if (err) {
-                    return cb(err);
-                }
-
-                cb(null, JSON.parse(body), response);
-            });
+            request("/api/deployments/", utils.createJsonCallback("listing deployments", cb));
         },
+
         get: function get(id, cb) {
-            request("/api/deployments/" + id, function deploymentGetCallback(err, response, body) {
-                if (err) {
-                    return cb(err);
-                }
+            var url = "/api/deployments/" + encodeURIComponent(id);
+            var action = "getting deployment with id " + id;
 
-                cb(null, JSON.parse(body), response);
-            });
+            request(url, utils.createJsonCallback(action, cb));
         },
+
         del: function del(id, cb) {
-            request.del("/api/deployments/" + id, function deploymentDelCallback(err, response, body) {
-                if (err) {
-                    return cb(err);
-                }
+            var url = "/api/deployments/" + encodeURIComponent(id);
+            var action = "deleting deployment with id " + id;
 
-                cb(null, body, response);
```

Comment:
The `body` is empty here (and in several other places) so I went with `createCallback`. Wanted to highlight it because altering the callback signature is a breaking change.